File: partitionList.js
function ListNode(val, next) {
    this.val = (val===undefined ? 0 : val)
    this.next = (next===undefined ? null : next)
}

var partition = function(head, x) {
    let lesser = new ListNode(0);
    let greater = new ListNode(0);
    let curr = head;
    head = lesser;
    let greaterHead = greater;
    while (curr != null) {
        if (curr.val < x) {
            lesser.next = curr;
            lesser = lesser.next;
        } else {
            greater.next = curr;
            greater = greater.next;
        }
        curr = curr.next;
        console.log(curr);
    }

    greater.next = null;
    lesser.next = greaterHead.next;
    return head.next;
};

File: partitionList.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def partition(self, head: Optional[ListNode], x: int) -> Optional[ListNode]:
        curr = head
        lesser = ListNode()
        greater = ListNode()
        greaterHead = greater
        head = lesser

        while curr:
            if curr.val < x:
                lesser.next = curr
                lesser = lesser.next
            else:
                greater.next = curr
                greater = greater.next
            curr = curr.next
        
        greater.next = None
        lesser.next = greaterHead.next

        return head.next